Ast/src/Lexer.cpp

```diff
@@ -535,37 +577,88 @@ Lexeme Lexer::readQuotedString()
             return Lexeme(Location(start, position()), Lexeme::BrokenString);
 
         case '\\':
+            readBackslashInString();
+            break;
+
+        default:
             consume();
-            switch (peekch())
-            {
-            case '\r':
-                consume();
-                if (peekch() == '\n')
-                    consume();
-                break;
+        }
+    }
 
-            case 0:
-                break;
+    consume();
 
-            case 'z':
-                consume();
-                while (isSpace(peekch()))
-                    consume();
-                break;
+    return Lexeme(Location(start, position()), Lexeme::QuotedString, &buffer[startOffset], offset - startOffset - 1);
+}
 
-            default:
-                consume();
-            }
+const Lexeme Lexer::nextInterpolatedString()
+{
+    Position start = position();
+    unsigned int startOffset = offset;
+
+    std::optional<Lexeme> readSectionOpt = readInterpolatedStringSection(start, Lexeme::InterpStringMid);
+
+    if (auto readSection = readSectionOpt)
```

Comment:
Why does this use an extra variable instead of checking readSectionOpt?

Reply:
Not 100% on what you have in mind, since I am not versed with std::optional in C++, but I believe I solved this

Reply:
Looks like what I had in mind is the same as what Alex posted